File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        c_s = Counter(s)
        print(c_s)
        for a in t:
            print(a)
            if a in c_s:
                if c_s[a] > 0:
                    c_s[a] = c_s[a] - 1
                else:
                    return False
            else:
                return False

        return True if len(set(c_s.values())) == 1 and list(set(c_s.values()))[0] == 0 else False

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        HashMap<String, Integer> sMap = new HashMap();
        for (int i = 0; i < s.length(); i++) {
            String key = String.valueOf(s.charAt(i));
            sMap.put(key, sMap.getOrDefault(key, 0) + 1);
        }

        for (int i = 0; i < t.length(); i++) {
            String letter = String.valueOf(t.charAt(i));
            if (!sMap.containsKey(letter)) return false;
            sMap.put(letter, sMap.get(letter) - 1);
            if (sMap.get(letter) == 0) sMap.remove(letter);
        }

        return true;

    }
}